tests/php/test-class-amp-youtube-embed-handler.php

```diff
@@ -188,6 +190,57 @@ public function test__conversion( $source, $expected, $fallback_for_expected = n
 		}
 	}
 
+	/**
+	 * Gets the test data for test_get_video_id_from_url().
+	 *
+	 * @return array The test data.
+	 */
+	public function get_video_id_data() {
+		return [
+			'basic_url'                        => [
+				'https://www.youtube.com/watch?v=XOY3ZUO6P0k',
+				'XOY3ZUO6P0k',
+			],
+			'short_url'                        => [
+				'https://youtu.be/XOY3ZUO6P0k',
+				'XOY3ZUO6P0k',
+			],
+			'url_with_underscore'              => [
+				'https://www.youtube.com/watch?v=CMrv_D78oxY',
+				'CMrv_D78oxY',
+			],
+			'short_url_with_underscore'        => [
+				'https://youtu.be/CMrv_D78oxY',
+				'CMrv_D78oxY',
+			],
+			'url_with_hyphen'                  => [
+				'https://www.youtube.com/watch?v=xo68-iWaKv8' . PHP_EOL,
+				'xo68-iWaKv8',
+			],
+			'url_with_hyphen_and_query_string' => [
+				'https://www.youtube.com/watch?v=xo68-iWaKv8&w=800&h=400' . PHP_EOL,
+				'xo68-iWaKv8',
+			],
```

Comment:
What about testing `/embed/` URLs?

Reply:
Sure, there are 2 tests added for URLs like that: https://github.com/ampproject/amp-wp/compare/b1344d537c456a926553396d2d90cc39c236f64b...f8ee4b9cb73434dcd60a819b5a73335fa98e00ef#diff-23071a193ac2ab7d9be3c9eb4bdebac6R228